src/Collection.php

```diff
@@ -562,6 +562,11 @@ public function intersperse($element, int $every = 1, int $startAt = 0): Collect
         return new self(Intersperse::of()($element)($every)($startAt), $this->getIterator());
     }
 
+    public function isEmpty(): bool
+    {
+        return !$this->getIterator()->valid();
+    }
```

Comment:
I got the idea for it from Laravel's LazyCollection: https://github.com/illuminate/collections/blob/master/LazyCollection.php#L556

Reply:
Yep, this is how it is supposed to be done.